src/components/NFTPage/Recommendation.vue

```diff
@@ -153,59 +238,24 @@ export default {
     swiper() {
       return this.$refs.recommendationSwiper.$swiper;
     },
-    nftFeaturedBooks() {
-      return this.bookstoreListItemForLandingPage.length
-        ? this.bookstoreListItemForLandingPage
-        : [...DEFAULT_RECOMMENDATIONS_LIST.BOOK.map(nft => ({ classId: nft }))];
-    },
-    normalizedList() {
-      const normalizedRecommendedList = [...this.recommendedList];
-      const displayItemCount = Number(this.displayItemCount);
-      if (this.recommendedList.length < displayItemCount) {
-        let index = 0;
-        let sourceData = this.isBookNft
-          ? this.nftFeaturedBooks
-          : this.nftFeaturedWNFT;
-        // TODO: remove filter after recommendation support collection
-        sourceData = sourceData.filter(nft =>
-          // TODO: find out and fix classId undefined reason
-          nft.classId?.startsWith('likenft1')
-        );
-        while (
-          normalizedRecommendedList.length < displayItemCount &&
-          index < sourceData.length
-        ) {
-          const itemToAdd = sourceData[index];
-          const alreadyExists = normalizedRecommendedList.some(
-            item => item.classId === itemToAdd.classId
-          );
-
-          if (!alreadyExists) {
-            normalizedRecommendedList.push(itemToAdd);
-          }
-          index += 1;
-        }
-      }
-      return normalizedRecommendedList;
-    },
   },
-  watch: {
-    async recommendedList(list) {
-      if (this.isBookNft && list.length < +this.displayItemCount) {
-        try {
-          await this.fetchBookstoreItemsFromCMSForLandingPage();
-        } catch (error) {
-          console.error(error);
-        }
+  mounted() {
+    if (this.isAuthorSpecific) {
+      this.lazyFetchCreatedNFTClassesByAddress(this.productOwner);
+    }
+    if (!this.isAuthorSpecific || !this.nftIsWritingNFT) {
```

Comment:
Should we use `type` here?

Reply:
[🎨 Use computed property for type & owner determination](https://github.com/likecoin/liker-land/pull/1863/commits/96ef0db4fb6b5a48870f8ec7cada2148278e5902)